config/generate_config.py

```diff
@@ -0,0 +1,792 @@
+# Copyright 2016-2024 Swiss National Supercomputing Centre (CSCS/ETH Zurich)
+# ReFrame Project Developers. See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+import re
+import sys
+import os
+import grp
+import subprocess
+import json
+import shlex
+import traceback
+from jinja2 import Environment, FileSystemLoader
+from utilities.constants import *
+from utils import *
+import socket
+import argparse
+
+system_config   = {}
+
+# Configure the logger
+logger = logging.getLogger()
+logger.setLevel(logging.DEBUG)
+console_handler = logging.StreamHandler()
+console_handler.setFormatter(CustomFormatter())
+logger.addHandler(console_handler)
+
+# Create an ArgumentParser object
+parser = argparse.ArgumentParser()
+
+# Define the '--auto' flag
+parser.add_argument('--auto', action='store_true', help='Turn off interactive mode')
+# Define the '--no-remote-containers' flag
+parser.add_argument('--no-remote-containers', action='store_true', help='Disable container platform detection in remote partition')
+# Define the '--no-remote-devices' flag
+parser.add_argument('--no-remote-devices', action='store_true', help='Disable devices detection in remote partition')
+# Define the '--reservations' flag
+parser.add_argument('--reservations', nargs='?', 
+    help='Specify the reservations that you want to create partitions for.'
+)
+
+args = parser.parse_args()
+
+user_input = not args.auto
+
+containers_search = 'y'
+if args.no_remote_containers:
+    containers_search = 'n'
+
+devices_search = 'y'
+if args.no_remote_devices:
+    devices_search = 'n'
+
+if args.reservations:
+    reservations_based = args.reservations.split(',')
+else:
+    reservations_based = False
+
+user_input = not args.auto
+
+def main(user_input, containers_search, devices_search, reservations_based):
+
+    system_config['systems']   = []
+
+    # Get the system name 
+    cluster_name = os.getenv('CLUSTER_NAME')
+    if cluster_name:
+        logger.info(f'System name is {cluster_name}\n')
+    else:
+        cluster_name = 'cluster'
+        logger.warning(f'System name not found set to "{cluster_name}\n"')
+
+    system_config['systems'].append({'name': cluster_name})
+
+    # Get the hostname
+    try:
+        hostname = socket.gethostname() # Same method reframe uses
+    except Exception as e:
+        hostname = '<hostname>'
+        logger.error(f'Hostname not found')
+        logger.debug(f'Trying to retrieve the hostname raised:\n{e}\n')
+    else:
+        hostname = hostname.strip()
+        hostname = re.search(r'(^\w+)', hostname)
+        hostname = hostname.group(0)
+        logger.info(f'Hostname is {hostname}\n')
+
+    if hostname != '<hostname>' and cluster_name != 'cluster':  
+        if hostname != cluster_name:
+            logger.warning(f'Detected hostname and ' +
+                          f'systemname are different\n') 

```

Comment:
```suggestion
            logger.warning('Detected hostname and '
                           'system name are different\n') 

```
No need for f-strings since you don't have any variable. You also don't need to add the strings since they are already inside a `()`.